code/config/lib/Loot/Config/Record.hs

```diff
@@ -83,20 +107,45 @@ type family Item' (k :: ConfigKind) (i :: ItemKind) where
     Item' 'Partial (l ::: t)  = Maybe t
     Item' 'Final   (l ::: t)  = t
     Item' k        (l ::< is) = ConfigRec k is
+    Item' 'Partial (l ::+ is) = ConfigRec 'Partial (SumSelection l : ToBranches is)
+    Item' 'Final   (l ::+ is) = ConfigRec 'Final   (SumSelection l : ToBranches is)
+    Item' 'Partial (l ::- is) = ConfigRec 'Partial is
+    Item' 'Final   (l ::- is) = Maybe (ConfigRec 'Final is)
```

Comment:
I started to write a long comment about how it would be great to just use an actual sum type instead of a product of `Maybe`s with saved `SumSelection` in final config, but then I realized that `vinyl` provides _only_ sum types (`Rec`s), and migrating the whole `loot-config` to `ntype` (https://github.com/int-index/ntype) would be quite an ordeal - almost the same ordeal as defining N-ary abstract sum type from scratch. So okay, this is a fine middle-ground.

Reply:
> `vinyl` provides _only_ sum types (`Rec`s)

you mean only *product* types right?

Anyway another advantage of doing it this way is that it is not disruptive and whoever was using `loot-config` before should not have any problem updating.

Reply:
>you mean only product types right?
Yes, that was a mistake.